### DISCUSSION TITLE
Best way to define variables in RAM

### DISCUSSION THREAD

**Discussion by rondnelson99** [vhelin/wla-dx#455] [Q&A] [answered]
I've been avidly using RGBDS for around a year now, but I wanted to try WLA for some other systems since it has a similar feature-set with its sections. I really like it so far, especially how arithmetic is always internally done with floats rather than integers. It really takes some hassle out of math. One thing I'm not too sure about is how to define variables that will reside in RAM. With RGBDS, I would create a section with type WRAM0 or WRAMX, and then I could use `db` or `ds` to reserve bytes in RAM. In this case, neither of those directives would require a fill value since the toolchain does not produce any value to fill RAM with. When I tried to do the same in WLA, it wanted to get fill values for those directives, indicating that they cannot be used to reserve RAM. So what is the intended approach to this problem? 
Thanks, 
Ron

**Comment by rondnelson99** [comment, marked as answer]
Oops, I just notices RAMSECTION in the docs. My bad. I'm sure that's what I need

**Comment by vhelin** [comment]
Yeah, I think .RAMSECTION is what you are looking for. We used to have only .ENUM in the past, but with .RAMSECTION you can "allocate" RAM in multiple RAM banks...
